flocker/node/_model.py

```diff
@@ -189,6 +164,39 @@ class Application(object):
     """
 
 
+@attributes(["dataset", "primary"])
+class Manifestation(object):
+    """
+    A dataset that is mounted on a node.
+
+    :ivar Dataset: The dataset being mounted.
+
+    :ivar bool primary: If true, this is a primary, otherwise it is a replica.
+    """
+
+
+@attributes(["dataset_id",
+             Attribute("maximum_size", default_value=None),
+             Attribute("metadata", default_factory=pmap)])
+class Dataset(object):
+    """
+    The filesystem data for a particular application.
+
+    :ivar dataset_id: A unique identifier, as ``unicode``. May also be ``None``
+        if this is coming out of human-supplied configuration, in which
+        case it will need to be looked up from actual state for existing
+        datasets, or a new one generated if a new dataset will need tbe
+        created.
+
+    :ivar PMap metadata: Mapping between ``unicode`` keys and
+        corresponding values. Typically there will be a ``"name"`` key whose
+        value is a a human-readable name, e.g. ``"main-postgres"``.
```

Comment:
If/when we open this up to arbitrary key/value pairs, we should be sure to reserve some of the key-space for ourselves.  For example, all upper-case identifiers or all identifiers beginning with `_` or whatever.  We both want to be able to expand the metadata Flocker itself provides without stomping on user-data and we don't want Flocker to unintentionally consume some metadata a user provides that just happens to use a key we picked too.

Probably nothing to do here now ... except make a note of this point somehow/somewhere?